lib/_emerge/depgraph.py

```diff
@@ -8625,7 +8633,7 @@ def _validate_blockers(self):
             dep_keys = Package._runtime_keys
             for myroot in self._frozen_config.trees:
                 if (
-                    self._frozen_config.myopts.get("--root-deps") is not None
+                    self._frozen_config.myopts.get("--root-deps") != "rdeps"
                     and myroot != self._frozen_config.target_root
```

Comment:
Looking back at e9740bcd93e01633695dee32425d1bab38e607c7, I'm not sure about the optimization logic. Once again, maybe we could try removing --root-deps from this expression entirely (basically revert e9740bcd93e01633695dee32425d1bab38e607c7).

It seems a9de7a29a25c876caab34b288cf87629514c7ac2 which was committed around the same time was also reverted at some point.

Reply:
I don't think keeping this does any harm, and I'd rather not break anything. Same for the above. I do intend to remove the `rdeps` option when we drop EAPI 6, although I was discussing the need for such an option that works with EAPI 7+ with someone today. `rdeps` is a terrible name, so I'd probably call it something else. I'm also considering other ways to solve their problem.

Reply:
Okay that seems fair to preserve backward compat. Should these conditions use `==` rather than `!=`, in order to affect the `--root-deps=rdeps` case, like this?
```diff
-               or self._frozen_config.myopts.get("--root-deps") is not None
+               or self._frozen_config.myopts.get("--root-deps") == "rdeps"
```

Reply:
Ah, good point. :upside_down_face: